In [1]:
# import libraries
import pandas as pd
import numpy as np
import os

In [2]:
ords_prod_merge = pd.read_pickle(r'C:\Users\New\Desktop\new\02 Data\Prepared Data\orders_products_merged.pkl')

In [3]:
df = ords_prod_merge[:1000000]

In [4]:
df.shape

(1000000, 16)

In [5]:
# Define fuction

def price_label(row):

    if row['prices'] <= 5:
        return 'Low range product'
    elif (row['prices'] > 5) and (row['prices'] <= 15):
        return 'Mid range product'
    elif row['prices'] > 15:
        return 'High range product'
    else: return 'Not enough data'

In [6]:
ords_prod_merge.head()

Unnamed: 0  product_id                product_name  aisle_id  \
0           0           1  Chocolate Sandwich Cookies        61   
1           0           1  Chocolate Sandwich Cookies        61   
2           0           1  Chocolate Sandwich Cookies        61   
3           0           1  Chocolate Sandwich Cookies        61   
4           0           1  Chocolate Sandwich Cookies        61   

   department_id  prices  order_id  user_id eval_set  order_number  \
0             19     5.8       -98     -118    prior            28   
1             19     5.8        47     -118    prior            30   
2             19     5.8       -37      -59    prior             2   
3             19     5.8       -30       -4    prior             1   
4             19     5.8       -52       -4    prior             3   

   orders_day_of_week  order_time_of_day  days_since_prior_order  \
0                   6                 11                     3.0   
1                   6                 17                    20.0   
2                   0                 21                     6.0   
3                   3                 13                     NaN   
4                   4                 17                     9.0   

   add_to_cart_order  reordered _merge  
0                  5          0   both  
1                  1          1   both  
2                 20          0   both  
3                 10          0   both  
4                 11          1   both

In [7]:
df['price_range'] = df.apply(price_label, axis=1)

C:\Users\New\AppData\Local\Temp\ipykernel_9928\2848743372.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['price_range'] = df.apply(price_label, axis=1)


In [9]:
df['price_range'].value_counts(dropna = False)

Mid range product     659373
Low range product     331283
High range product      9344
Name: price_range, dtype: int64

In [10]:
df['prices'].max()

24.5

In [14]:
df.loc[df['prices'] > 15, 'price_range_loc'] = 'High range product'

In [12]:
df.loc[(df['prices'] <= 15) & (df['prices'] > 5), 'price_range_loc'] = 'Mid-range product'

In [13]:
 df.loc[df['prices'] <= 5, 'price_range_loc'] = 'Low-range product'

In [15]:
df['price_range'].value_counts(dropna = False)

Mid range product     659373
Low range product     331283
High range product      9344
Name: price_range, dtype: int64

In [17]:
ords_prod_merge['orders_day_of_week'].value_counts(dropna = False)

0    6209632
1    5665830
6    4500246
2    4217766
5    4209449
3    3844096
4    3787193
Name: orders_day_of_week, dtype: int64

In [40]:
result = []

for value in ords_prod_merge["orders_day_of_week"]:
    if value == 0:
        result.append("Busiest day")
    elif value == 4:
        result.append("Least busy")
    else:
        result.append("Regularly busy")

In [41]:
result

['Regularly busy',
 'Regularly busy',
 'Busiest day',
 'Regularly busy',
 'Least busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Least busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Busiest day',
 'Busiest day',
 'Busiest day',
 'Busiest day',
 'Busiest day',
 'Least busy',
 'Regularly busy',
 'Regularly busy',
 'Busiest day',
 'Regularly busy',
 'Regularly busy',
 'Least busy',
 'Regularly busy',
 'Busiest day',
 'Regularly busy',
 'Busiest day',
 'Least busy',
 'Busiest day',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Bus

In [21]:
ords_prod_merge['busiest_day'] = result

In [39]:
ords_prod_merge['busiest_day'].value_counts()

Regularly busy    32434212
Name: busiest_day, dtype: int64

In [42]:
result_2 = []

for value in ords_prod_merge["orders_day_of_week"]:
  if value == [0,1]:
    result_2.append("Busiest days")
  elif value == [3,4]:
    result_2.append("Least busy")
  else:
    result_2.append("Regularly busy")

In [43]:
result_2

['Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly busy',
 'Regularly 

In [44]:
ords_prod_merge['busiest_days'] = result_2

In [45]:
ords_prod_merge['busiest_days'].value_counts()

Regularly busy    32434212
Name: busiest_days, dtype: int64

In [33]:
# creating new column 'busiest_days' with the results of the loop.

ords_prod_merge['busiest_days'] = result_2

In [46]:
ords_prod_merge.head()

Unnamed: 0  product_id                product_name  aisle_id  \
0           0           1  Chocolate Sandwich Cookies        61   
1           0           1  Chocolate Sandwich Cookies        61   
2           0           1  Chocolate Sandwich Cookies        61   
3           0           1  Chocolate Sandwich Cookies        61   
4           0           1  Chocolate Sandwich Cookies        61   

   department_id  prices  order_id  user_id eval_set  order_number  \
0             19     5.8       -98     -118    prior            28   
1             19     5.8        47     -118    prior            30   
2             19     5.8       -37      -59    prior             2   
3             19     5.8       -30       -4    prior             1   
4             19     5.8       -52       -4    prior             3   

   orders_day_of_week  order_time_of_day  days_since_prior_order  \
0                   6                 11                     3.0   
1                   6                 17                    20.0   
2                   0                 21                     6.0   
3                   3                 13                     NaN   
4                   4                 17                     9.0   

   add_to_cart_order  reordered _merge     busiest_day    busiest days  \
0                  5          0   both  Regularly busy  Regularly busy   
1                  1          1   both  Regularly busy  Regularly busy   
2                 20          0   both  Regularly busy     Busiest day   
3                 10          0   both  Regularly busy  Regularly busy   
4                 11          1   both  Regularly busy      Least busy   

     busiest_days  
0  Regularly busy  
1  Regularly busy  
2  Regularly busy  
3  Regularly busy  
4  Regularly busy

In [55]:
ords_prod_merge['orders_day_of_week'].value_counts()

0    6209632
1    5665830
6    4500246
2    4217766
5    4209449
3    3844096
4    3787193
Name: orders_day_of_week, dtype: int64

# The sum of days 0 & 1

busiest_days 6209632 + 5665830 = 11875462

# The sum of days 3 & 4

slowest_days 3844096 + 3787193 = 7631289


In [57]:
ords_prod_merge['order_time_of_day'].value_counts()

10    2764390
11    2738585
14    2691563
15    2664522
13    2663272
12    2620800
16    2537469
9     2456661
17    2089452
8     1719952
18    1637922
19    1259382
20     977017
7      891928
21     796362
22     634737
23     402612
6      290770
0      218942
1      115786
5       88057
2       69431
4       53283
3       51317
Name: order_time_of_day, dtype: int64

In [60]:
result_3 = []

for value in ords_prod_merge["order_time_of_day"]:
    if value>=9 and value<=16:
        result_3.append("Busiest day")
    elif value == 4:
        result_3.append("Least busy")
    else:
        result_3.append("Regularly busy")

In [61]:
# Creating busiest period of day

ords_prod_merge['busiest_period_of_day'] = result_3

In [62]:
ords_prod_merge.head()

Unnamed: 0  product_id                product_name  aisle_id  \
0           0           1  Chocolate Sandwich Cookies        61   
1           0           1  Chocolate Sandwich Cookies        61   
2           0           1  Chocolate Sandwich Cookies        61   
3           0           1  Chocolate Sandwich Cookies        61   
4           0           1  Chocolate Sandwich Cookies        61   

   department_id  prices  order_id  user_id eval_set  order_number  \
0             19     5.8       -98     -118    prior            28   
1             19     5.8        47     -118    prior            30   
2             19     5.8       -37      -59    prior             2   
3             19     5.8       -30       -4    prior             1   
4             19     5.8       -52       -4    prior             3   

   orders_day_of_week  order_time_of_day  days_since_prior_order  \
0                   6                 11                     3.0   
1                   6                 17                    20.0   
2                   0                 21                     6.0   
3                   3                 13                     NaN   
4                   4                 17                     9.0   

   add_to_cart_order  reordered _merge     busiest_day    busiest days  \
0                  5          0   both  Regularly busy  Regularly busy   
1                  1          1   both  Regularly busy  Regularly busy   
2                 20          0   both  Regularly busy     Busiest day   
3                 10          0   both  Regularly busy  Regularly busy   
4                 11          1   both  Regularly busy      Least busy   

     busiest_days busiest_period_of_day  
0  Regularly busy           Busiest day  
1  Regularly busy        Regularly busy  
2  Regularly busy        Regularly busy  
3  Regularly busy           Busiest day  
4  Regularly busy        Regularly busy

In [63]:
ords_prod_merge['busiest_period_of_day'].value_counts()

Busiest day       21137262
Regularly busy    11243667
Least busy           53283
Name: busiest_period_of_day, dtype: int64

In [64]:
path = r'C:\Users\New\Desktop\new'

In [65]:
ords_prod_merge.to_pickle(os.path.join(path, '02 Data','Prepared Data', 'orders__products_merge_new.pkl'))